e2e/nomostest/registryproviders/artifact_registry.go

```diff
@@ -121,33 +121,17 @@ type ArtifactRegistryOCIProvider struct {
 
 // Login to the registry with the crane client
 func (a *ArtifactRegistryOCIProvider) Login() error {
-	var err error
-	authCmd := a.shell.Command("gcloud", "auth", "print-access-token")
-	loginCmd := a.shell.Command("crane", "auth", "login",
-		"-u", "oauth2accesstoken", "--password-stdin",
-		a.registryHost())
-	loginCmd.Stdin, err = authCmd.StdoutPipe()
-	if err != nil {
-		return fmt.Errorf("creating STDOUT pipe: %w", err)
-	}
-	if err := loginCmd.Start(); err != nil {
-		return fmt.Errorf("starting login command: %w", err)
-	}
-	if err := authCmd.Run(); err != nil {
-		return fmt.Errorf("running print-access-token command: %w", err)
-	}
-	if err := loginCmd.Wait(); err != nil {
-		return fmt.Errorf("waiting for login command: %w", err)
+	// Docker config used by crane client
+	if err := a.shell.GcloudDockerLogin(a.registryHost()); err != nil {
+		return fmt.Errorf("oci provider logging into artifact registry: %w", err)
 	}
 	return nil
 }
 
 // Logout of the registry with the crane client
 func (a *ArtifactRegistryOCIProvider) Logout() error {
-	_, err := a.shell.ExecWithDebug("crane", "auth", "logout", a.registryHost())
-	if err != nil {
-		return fmt.Errorf("running logout command: %w", err)
-	}
+	// Never log out of docker clients.
```

Comment:
Should we just remove the logout methods/interfaces?

Reply:
Assuming we dont need to log into the local registries, I'll just move the "login" to the env setup.

Reply:
Oh wait. I can't. It depends on the location, which is a provider field...

Reply:
I think you should be able to move login back to the `Setup` method so that it's done once at the beginning of the test suite. The provider is scoped to the test suite rather than per testcase

Reply:
I'm gonna add client configs to NT and just make them all independent per environment under TmpDir so we dont have to worry about them conflicting

Reply:
Login/Logout has been refactored. Please review. I think they can be left in the new refactor because they have comments that explain why it's not needed, and the caller doesn't need to know.